test/services/katello/registration_manager_test.rb

```diff
@@ -91,6 +91,19 @@ def test_host_profile_assume_build_mode_only_in_build
           assert @klass.validate_hosts(hosts, @org, @host.name, 'different-uuid')
         end
 
+        def test_re_register_build_mode
```

Comment:
These adjustment got the test working as expected for me:

```
[vagrant@centos7-katello-devel katello]$ git diff
diff --git a/test/services/katello/registration_manager_test.rb b/test/services/katello/registration_manager_test.rb
index e6bc4c3..b72819d 100644
--- a/test/services/katello/registration_manager_test.rb
+++ b/test/services/katello/registration_manager_test.rb
@@ -94,13 +94,13 @@ module Katello
         def test_re_register_build_mode
           @host = FactoryBot.create(:host, :with_subscription, :managed, organization: @org)
           @host.subscription_facet.update(dmi_uuid: 'existing_system_uuid')
-          @host.build = false
+          refute @host.build
           Setting[:host_re_register_build_only] = true
 
-          error = assert_raises(Katello::Errors::RegistrationError) { @klass.validate_hosts(hosts, @org, @host.name) }
+          error = assert_raises(Katello::Errors::RegistrationError) { @klass.validate_hosts(hosts, @org, @host.name, nil) }
           assert_match(/currently registered/, error.message)
 
-          @host.build = true
+          @host.update(build: true)
           assert @klass.validate_hosts(hosts, @org, @host.name, 'existing_system_uuid')
         end

```

Reply:
I think I've pushed a fix with the changes applied correctly.  My local testing is all messed up and everything fails.....